astropy/modeling/tabular.py

```diff
@@ -92,40 +95,51 @@ class _Tabular(Model):
 
     _id = 0
 
-    def __init__(self, points=None, lookup_table=None, method='linear',
-                 bounds_error=True, fill_value=np.nan, **kwargs):
+    def __init__(self, points=None, lookup_table=None,
+                 method='linear', bounds_error=True, fill_value=np.nan,
+                 **kwargs):
 
         n_models = kwargs.get('n_models', 1)
         if n_models > 1:
             raise NotImplementedError('Only n_models=1 is supported.')
         super(_Tabular, self).__init__(**kwargs)
 
-        if lookup_table is not None:
+        if lookup_table is None:
+            raise ValueError('Must provide a lookup table.')
+
+        if not isinstance(lookup_table, u.Quantity):
             lookup_table = np.asarray(lookup_table)
-            if self.lookup_table.ndim != lookup_table.ndim:
```

Comment:
Note: This comparison does not make sense to me and complicate things after my changes, so I removed it. Please let me know if there is a reason to keep this.

Reply:
The reason is that this is a dynamic class which serves as a base class in `tabular_model` and allows for creating tabular models with any dimension. A specific example:
```
tabular_3d_class = tabular_model(dim=3) # create the equivalent of Tabular3D
tinstance = tabular_3d_class(lookup_table=[1,2,3,4]) # this fails because it expects a 3D lookup table
```

Basically the function creates a class with the correct members for the dimensionality.

Reply:
OK I added it back in. Please review.